src/lib/utils/twMergeConfig.tsx

```diff
@@ -0,0 +1,18 @@
+import {extendTailwindMerge} from "tailwind-merge"
+
+// Creates an array containing type-# classes from 0 to 1
+const typeClasses = Array.from({length: 10}, (_, i) => `type-${i}`)
+// Creates an array containing text-# classes from 11 to 30
+const textClasses = Array.from({length: 20}, (_, i) => `text-${i + 11}`)
+
+const typographyClasses = [...textClasses, ...typeClasses]
+
+const twMerge = extendTailwindMerge({
+  extend: {
+    classGroups: {
+      "font-size": typographyClasses,
```

Comment:
This doesn't resolve our unique heading situation since something like 
```
type-3 xl:text-30 type-2
```
would result in `xl:text-30 type-2` instead of `type-2` since we're making specific heading changes for Press. But generally, it should work ok when it's `text-#` vs `type-#` (e.g. twMerge will use the last defined font-size type) 😎

Reply:
I ran a couple senarios through the custom function and it does seem to respect the desired classes. Here's the output I tried

```
custom twMerge("text-18 type-2 xl:text-18 xl:type-2"): type-2 xl:type-2
custom twMerge("type-2 text-18 xl:type-2 xl:text-18"): text-18 xl:text-18
twMerge("text-18 type-2 xl:text-18 xl:type-2"): text-18 type-2 xl:text-18 xl:type-2
twMerge("type-2 text-18 xl:type-2 xl:text-18"): type-2 text-18 xl:type-2 xl:text-18
```